src/libs/merge/kdbmerge.c

```diff
@@ -988,21 +988,13 @@ KeySet * elektraMerge (KeySet * our, Key * ourRoot, KeySet * their, Key * theirR
 	}
 
 #ifdef LIBGITFOUND
-#ifndef CMERGE_ON_LINUX
 	git_libgit2_init ();
-	ELEKTRA_LOG ("Initializing LibGit2");
-#else
-	ELEKTRA_LOG ("Not initializing LibGit2, but using it all the same.");
-#endif
 	ELEKTRA_LOG ("cmerge can use libgit2 to handle arrays");
 	if (handleArrays (ourCropped, theirCropped, baseCropped, result, informationKey, strategy) > 0)
 	{
 		ksDel (result);
 		return NULL;
 	}
-#ifndef CMERGE_ON_LINUX
-	git_libgit2_shutdown ();
```

Comment:
Why do we do the init but no shutdown here?
Btw if it is definitely an upstream problem/leak we can make a valgrind suppression for it and report the issue there.

Reply:
Some libraries are not safe to be initialized again after shutdown. Furthermore we do not really know if anybody else is using libgit2 in our process. So it can make sense to not call shutdown (if it is not working properly).

The whole global init and shutdown is calling for a lot of trouble. Unfortunately, many libraries do not use proper solutions (like handles).

In theory (with mutex protection and ref counting) multiple init/shutdown could be supported but it is rare that this is working properly.

> and report the issue there

I agree.

Reply:
> Some libraries are not safe to be initialized again after shutdown.

Ok, but the shutdown was clearly called before, and now we only call init. I assumed it was removed by mistake. If there is a reason not to call it, I'm fine with that!

Reply:
> Why do we do the init but no shutdown here?

That was a mistake :/ Nonetheless, it is actually optional

> Usually you don’t need to call the shutdown function as the operating system will take care of reclaiming resources, but if your application uses libgit2 in some areas which are not usually active, you can use git_libgit2_shutdown(); to ask the library to clean up the global state. ([source](https://libgit2.org/docs/guides/101-samples/))

---

> Btw if it is definitely an upstream problem/leak we can make a valgrind suppression for it and report the issue there.

The person who answered my  [stackoverflow question](https://stackoverflow.com/questions/58477412/memory-problems-with-libgit2-initialization) also appears in the LibGit2 contributor list, so I assume they are aware.

Reply:
> I added memleak labels for both tests in this PR.

It is better if we suppress everything from libgit. This way we still check if our code has memleaks.

>   The person who answered my stackoverflow question also appears in the LibGit2 contributor list, so I assume they are aware.

You never know. Better we report it. But they are obviously only interested in the latest versions, maybe they already fixed it?